### ISSUE TITLE
Run prettier on CI instead of precommit hook

### ISSUE THREAD

**Issue by obartra** [obartra/reflex#48]
`prettier` and `eslint --fix` should run on CI and, if there are any changes, amend the commit. This will cancel the current build and trigger a new one but I think it's an ok tradeoff (it shouldn't happen when using VSC with the prettier plugin, the format task can also be run manually and the CI build would be quick to rebuild)

This would make (formatted) commits faster and also enforce formatting in CI (while preventing inconsistencies from `--no-verify`).

It would also remove dependencies (husky, lint-staged)

**Comment by obartra** [comment]
I don't care too much about this one and I'm afraid it will just be trading one can of worms for another. Since the linting task is already running anyway, are you ok with us closing this one @arahansen ?

**Comment by arahansen** [comment]
What sort of can of worms would we be introducing to CI? Just the additional overhead of amending commits?

**Comment by obartra** [comment]
I think the main one is just when amending the commit retriggering an additional CI build, it's not that bad because it shouldn't happen if we keep husky and lint-staged.

If we keept it though then we are at exactly where we are today. If we remove it, then every CI build will effectively trigger 2 CI builds unless we skip the second one.

To skip the second one we would require changing to a CI blacklisted branch but that would limit concurrent builds, require a new PR...

TLDR: some worms, not worth imo

**Comment by obartra** [comment]
Closing for now, we can reopen if needed
